### PR TITLE
tests/robustness: Unify model test cases

### PR DESCRIPTION
**Pull request by serathius**

### CODE REVIEW

File: tests/robustness/model/non_deterministic_test.go [hunk 1]
```diff
@@ -19,116 +19,18 @@ import (
...
-			operations: []testOperation{
-				{req: compareRevisionAndPutRequest("key", 0, "42"), resp: compareRevisionAndPutResponse(false, 42)},
-			},
-		},
-		{
-			name: "Get response data should match put",
-			operations: []testOperation{
-				{req: putRequest("key1", "11"), resp: putResponse(1)},
-				{req: putRequest("key2", "12"), resp: putResponse(2)},
-				{req: getRequest("key1"), resp: getResponse("key1", "11", 1, 1), failure: true},
```
Comment (fuweid) [inline tests/robustness/model/non_deterministic_test.go:80]: I think it's caused by that `toNonDeterministicTest` changes failure from `true` to `false`.

File: tests/robustness/model/non_deterministic_test.go [hunk 2]
```diff
@@ -648,6 +331,32 @@ func TestModelNonDeterministic(t *testing.T) {
...
 
+type nonDeterministicModelTest struct {
+	name       string
+	operations []nonDeterministicOperation
+}
+
+type nonDeterministicOperation struct {
+	req     EtcdRequest
+	resp    EtcdNonDeterministicResponse
+	failure bool
```
Comment (ahrtr) [inline tests/robustness/model/non_deterministic_test.go:342]:
```suggestion
	expectFailure bool
```

File: tests/robustness/model/deterministic_test.go [hunk 3]
```diff
@@ -20,378 +20,382 @@ import (
...
 )
 
-func TestModelBase(t *testing.T) {
-	type testOperation struct {
-		req     EtcdRequest
-		resp    EtcdResponse
-		failure bool
+func TestModelDeterministic(t *testing.T) {
+	for _, tc := range deterministicModelTestScenarios {
+		t.Run(tc.name, func(t *testing.T) {
```
Comment (ahrtr) [inline tests/robustness/model/deterministic_test.go:25]: nit, and the same comment for the nonDeterministicTestScenarios.
```suggestion
		tc := tc
		t.Run(tc.name, func(t *testing.T) {
```

**Review by ahrtr — APPROVED** [review]
LGTM with minor comments.

File: tests/robustness/model/non_deterministic_test.go (hunk 2 above)
Comment (serathius) [reply tests/robustness/model/non_deterministic_test.go:342]: Done

File: tests/robustness/model/deterministic_test.go (hunk 3 above)
Comment (serathius) [reply tests/robustness/model/deterministic_test.go:25]: Done

**Pull request merged**